### DISCUSSION TITLE
Trackpad Camera Control

### DISCUSSION THREAD

**Discussion by puprs** [runelite/runelite#17061] [Ideas]
Hello,

I'd love a plugin to be made to change the way camera is controlled with a trackpad, specifically in my case on a mac.

Currently you can either move the camera with arrow keys, or use the settings/plugin to allow you to move camera with right click.
Both of these control methods are a bit cumbersome.

The plugin would be added to 'mimic' how camera movement exists in things like Blender with a trackpad.
In that instance you are able to move the camera around (panning) with 2 fingers on the trackpad without clicking.
You are then able to zoom in and out with the 'pinch' gesture.

This camera control makes a lot more sense and is a lot more intuitive than currently available within Runelite.

I have attached a video file to highlight what I'm meaning.

https://github.com/runelite/runelite/assets/89594591/a2a77e12-6066-4b32-8954-b1f0da271ef1

**Comment by crichmond1989** [comment]
https://github.com/runelite/plugin-hub/pull/10820

**Reply by crichmond1989** [reply]
I attempted everything you requested in the video, but the pinch gesture is lost in translation somewhere between MacOS -> Java -> Runelite. I was able to do swipe left/right for yaw (horizontal), and CTRL + swipe up/down for pitch. Regular swipe up/down comes across as a scroll event and is handled natively by OSRS.

I have an "Invert" setting for each of these interactions. I personally keep the default MacOS scroll direction on trackpad, but wanted to invert that direction for OSRS zoom.

This works on my laptop and I'm going through the Runelite plugin publish process for the first time. I'll update when it's been approved.
